c/driver_manager/adbc_driver_manager.cc

```diff
@@ -15,26 +15,45 @@
 // specific language governing permissions and limitations
 // under the License.
 
-#include "arrow-adbc/adbc_driver_manager.h"
+#if defined(_WIN32)
+#include <windows.h>  // Must come first
```

Comment:
should we define _NOMINMAX too just to be safe?

Reply:
When I google `_NOMINMAX` it looks like the pre-processor macro doesn't have the leading underscore in the results. Is it supposed to have the underscore or not?

Also, I probably wouldn't want to add the define unless we actually do encounter the issue and need it personally. Thoughts?